bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/InsightParser.java

```diff
@@ -0,0 +1,150 @@
+    private static final int INSIGHT_POSITION_TIMESPAN = 5;
+    private static final int INSIGHT_POSITION_AVERAGEPOWER = 6;
+    private static final int INSIGHT_POSITION_CURRENTPOWER = 7;
+    private static final int INSIGHT_POSITION_ENERGYTODAY = 8;
+    private static final int INSIGHT_POSITION_ENERGYTOTAL = 9;
+    private static final int INSIGHT_POSITION_STANDBYLIMIT = 10;
+
+    private final String value;
+
+    public InsightParser(String value) {
+        this.value = value;
+    }
+
+    /**
+     * Parse provided string of values.
+     *
+     * @return Map of channel id's with states
+     */
+    public Map<String, State> parse() {
+        HashMap<String, State> result = new HashMap<>();
+        String[] params = value.split("\\|");
+        for (int i = 0; i < params.length; i++) {
+            String value = params[i];
+            switch (i) {
+                case INSIGHT_POSITION_STATE:
+                    result.put(WemoBindingConstants.CHANNEL_STATE, getOnOff(value));
+                    break;
+                case INSIGHT_POSITION_LASTCHANGEDAT:
+                    result.put(WemoBindingConstants.CHANNEL_LASTCHANGEDAT, getDateTime(value));
+                    break;
+                case INSIGHT_POSITION_LASTONFOR:
+                    result.put(WemoBindingConstants.CHANNEL_LASTONFOR, getNumber(value));
+                    break;
+                case INSIGHT_POSITION_ONTODAY:
+                    result.put(WemoBindingConstants.CHANNEL_ONTODAY, getNumber(value));
+                    break;
+                case INSIGHT_POSITION_ONTOTAL:
+                    result.put(WemoBindingConstants.CHANNEL_ONTOTAL, getNumber(value));
+                    break;
+                case INSIGHT_POSITION_TIMESPAN:
+                    result.put(WemoBindingConstants.CHANNEL_TIMESPAN, getNumber(value));
+                    break;
+                case INSIGHT_POSITION_AVERAGEPOWER:
+                    result.put(WemoBindingConstants.CHANNEL_AVERAGEPOWER, getPowerFromWatt(value));
+                    break;
+                case INSIGHT_POSITION_CURRENTPOWER:
+                    result.put(WemoBindingConstants.CHANNEL_CURRENTPOWER, getPowerFromMilliWatt(value));
+                    break;
+                case INSIGHT_POSITION_ENERGYTODAY:
+                    result.put(WemoBindingConstants.CHANNEL_ENERGYTODAY, getEnergy(value));
+                    break;
+                case INSIGHT_POSITION_ENERGYTOTAL:
+                    result.put(WemoBindingConstants.CHANNEL_ENERGYTOTAL, getEnergy(value));
+                    break;
+                case INSIGHT_POSITION_STANDBYLIMIT:
+                    result.put(WemoBindingConstants.CHANNEL_STANDBYLIMIT, getPowerFromMilliWatt(value));
+                    break;
+            }
+        }
+        return result;
+    }
+
+    private State getOnOff(String value) {
+        return "0".equals(value) ? OnOffType.OFF : OnOffType.ON;
```

Comment:
You prefer that option to OnOffType.from(value) ?
It could make sense if the value could have other value than 0 and 1.

Reply:
I don't prefer it, but unfortunately it can be non-binary. When receiving the parameters from a HTTP call the value can be 8 for ON (just verified by testing). I have added a test case for this.

Reply:
Ok, that is a good argument.